earthvision/datasets/cowc.py

```diff
@@ -1,2 +1,119 @@
+"""Cars Overhead with Context."""
+import os
+import shutil
+import posixpath
+import tarfile
+import torch
+import numpy as np
+import pandas as pd
+from torch.utils.data import Dataset
+from .utils import _urlretrieve, _load_img
+from ..constants.COWC.config import file_mapping_counting, \
+                                    file_mapping_detection
+
+
 class COWC():
-    raise NotImplementedError
+    """ Cars Overhead with Context.
+    https://gdo152.llnl.gov/cowc/
+    """
+
+    mirrors = "https://storage.googleapis.com/ossjr"
+    resources = "cowc-everything-small.txz"
+
+    def __init__(self,
+                 root: str,
+                 data_mode: str = 'train',
+                 task_mode: str = 'counting',
+                 transform=None,
+                 target_transform=None):
+        self.root = root
+        self.data_mode = data_mode
+        self.task_mode = task_mode
+        self.transform = transform
+        self.target_transform = target_transform
+
+        if not self._check_exists():
+            self.download()
+            self.extract_file()
+
+        if self.task_mode == 'counting':
+            self.task_path = os.path.join(
+                self.root, 'cowc/datasets/patch_sets/counting'
+            )
+            self.file_mapping = file_mapping_counting
+        else: # self.task_mode = 'detection'
+            self.task_path = os.path.join(
+                self.root, 'cowc/datasets/patch_sets/detection'
+            )
+            self.file_mapping = file_mapping_detection
+            
+        for filename, compressed in self.file_mapping.items():
+            if not self._check_exists_subfile(filename):
+                self.extract_subfile(filename, compressed)
+            
+        self.img_labels = self.get_path_and_label()
+
+    def __getitem__(self, idx):
+        img_path = self.img_labels.iloc[idx, 0]
+        label = self.img_labels.iloc[idx, 1]
+        folder = img_path.split('/', 1)[0]
+        img_path = os.path.join(self.task_path, folder, img_path)
+        image = _load_img(img_path)
+        
+        if self.transform:
+            image = self.transform(image)
+        if self.target_transform:
+            label = self.target_transform(label)
+        
+        image = np.array(image)
+        image = torch.from_numpy(image)
+        sample = (image, label)
+
+        return sample
+
+    def __len__(self):
+        return len(self.img_labels)
+
+    def get_path_and_label(self):
+        """Return dataframe type consist of image path 
+        and corresponding label."""
+        
+        if self.task_mode == 'counting':
+            if self.data_mode == 'train':
+                label_name = 'COWC_train_list_64_class.txt.bz2'
+            else: # self.data_mode == 'test'
```

Comment:
@andreaschandra @dwiuzila 
kalo else doang nnti yg input `data_mode = 'ayam'` gimana? meningan `elif self.data_mode == 'test'`,
terus tambahin `else: 'data_mode not recognized'`